perses/rjmc/geometry.py

```diff
@@ -293,6 +293,8 @@ def _logp_propose(self, top_proposal, old_positions, beta, new_positions=None, d
             else:
                 if direction == 'forward':
                     constraint = self._get_bond_constraint(atom, bond_atom, top_proposal.new_system)
+                    if constraint is None:
```

Comment:
I don't think that should be necessary. I don't see the need to type check the return value of every function.